### ISSUE TITLE
Can we generate attacks for multiple-output models.? 

### ISSUE THREAD

**Issue by ahmedomer6** [IBM/adversarial-robustness-toolbox#430]
I am building multiple output classifier and I want to generate attacks for that. I tried, but it doesn't work since the ART classifier's predict function has one result array with two dims <#samples, predictions>

Do I miss an existing feature of ART.? Is there any way in the ART that let me do such job.?

**Comment by beat-buesser** [comment]
Hi @ahmedomer6 Most of the features of ART so far are focusing on single output models. Could you please provide more details about how you are attacking the multiple outputs of your model?

**Comment by ahmedomer6** [comment]
Hi @beat-buesser, I am trying to figure it out since there are multiple gradients and hence attack for each. My intial thought is to let the ART classifiers choose a specific output/layer for generating the attacks.

**Comment by beat-buesser** [comment]
@Hi @ahmedomer6 Do you plan to attack one single output out of the multiple outputs of your model? In this case you could use `KerasClassifier` and its argument `output_layer` which accepts the index of the output to be used in the attack. The same is available with `input_layer` for multiple input layers to define the input where loss gradients should be calculated.

**Comment by ahmedomer6** [comment]
hi @beat-buesser many thanks for your advice. I know this feature, but I have doubt that it works with Functional models in which there are shared layers. Besides, `KerasClassifier` not flexible with custom loss functions.
